3.1_parameters.ipynb

```diff
@@ -213,70 +214,65 @@
     "  }\n",
     "}\n",
     "\n",
-    "// verify the inputs are sorted\n",
-    "class Sort4AscendingTester(c: Sort4) extends PeekPokeTester(c) {\n",
-    "  poke(c.io.in0, 3)\n",
-    "  poke(c.io.in1, 6)\n",
-    "  poke(c.io.in2, 9)\n",
-    "  poke(c.io.in3, 12)\n",
-    "  expect(c.io.out0, 3)\n",
-    "  expect(c.io.out1, 6)\n",
-    "  expect(c.io.out2, 9)\n",
-    "  expect(c.io.out3, 12)\n",
-    "\n",
-    "  poke(c.io.in0, 13)\n",
-    "  poke(c.io.in1, 4)\n",
-    "  poke(c.io.in2, 6)\n",
-    "  poke(c.io.in3, 1)\n",
-    "  expect(c.io.out0, 1)\n",
-    "  expect(c.io.out1, 4)\n",
-    "  expect(c.io.out2, 6)\n",
-    "  expect(c.io.out3, 13)\n",
-    "\n",
-    "  poke(c.io.in0, 13)\n",
-    "  poke(c.io.in1, 6)\n",
-    "  poke(c.io.in2, 4)\n",
-    "  poke(c.io.in3, 1)\n",
-    "  expect(c.io.out0, 1)\n",
-    "  expect(c.io.out1, 4)\n",
-    "  expect(c.io.out2, 6)\n",
-    "  expect(c.io.out3, 13)\n",
     "\n",
-    "}\n",
-    "class Sort4DescendingTester(c: Sort4) extends PeekPokeTester(c) {\n",
-    "  poke(c.io.in0, 3)\n",
-    "  poke(c.io.in1, 6)\n",
-    "  poke(c.io.in2, 9)\n",
-    "  poke(c.io.in3, 12)\n",
-    "  expect(c.io.out0, 12)\n",
-    "  expect(c.io.out1, 9)\n",
-    "  expect(c.io.out2, 6)\n",
-    "  expect(c.io.out3, 3)\n",
-    "\n",
-    "  poke(c.io.in0, 13)\n",
-    "  poke(c.io.in1, 4)\n",
-    "  poke(c.io.in2, 6)\n",
-    "  poke(c.io.in3, 1)\n",
-    "  expect(c.io.out0, 13)\n",
-    "  expect(c.io.out1, 6)\n",
-    "  expect(c.io.out2, 4)\n",
-    "  expect(c.io.out3, 1)\n",
-    "    \n",
-    "  poke(c.io.in0, 1)\n",
-    "  poke(c.io.in1, 6)\n",
-    "  poke(c.io.in2, 4)\n",
-    "  poke(c.io.in3, 13)\n",
-    "  expect(c.io.out0, 13)\n",
-    "  expect(c.io.out1, 6)\n",
-    "  expect(c.io.out2, 4)\n",
-    "  expect(c.io.out3, 1)\n",
-    "\n",
-    "}\n",
     "\n",
     "// Here are the testers\n",
-    "val worksAscending = iotesters.Driver(() => new Sort4(true)) { c => new Sort4AscendingTester(c) }\n",
-    "val worksDescending = iotesters.Driver(() => new Sort4(false)) { c => new Sort4DescendingTester(c) }\n",
-    "assert(worksAscending && worksDescending) // Scala Code: if works == false, will throw an error\n",
+    "test(new Sort4(true)) { c => \n",
+    "  c.io.in0.poke(3.U)\n",
```

Comment:
Might also be a prime candidate for a test helper function to reduce code length

Reply:
Agreed, it will have to be later.

Reply:
I have created Issue #131 to remember this